### ISSUE TITLE
Content Type not enforced in s3 pre-signed POST

### ISSUE THREAD

**Issue by tglines** [aws/aws-sdk-js#2212]
I'm trying to figure out content-type enforcement for s3 presigned post's.  

I'm currently using the AWS javascript SDK to create pre-signed POST's. (note that this question is NOT about pre-signed PUTs/URLs, another s3 feature)

When trying to create a pre-signed post I do something like the following:

```
const params = {
  Bucket: 'myuniques3bucket',
  Fields: {
    Key: 'key1.png',
    'Content-Type': 'image/png'
  },
  conditions: [
    {bucket: 'myuniques3bucket'},
    {key: 'key1.png'},
    {'Content-Type': 'image/png'},
    ['content-length-range', 1024, 1048576], // 1KB to 10MB
    {'x-amz-date': amzDate},
    {'x-amz-algorithm': 'AWS4-HMAC-SHA256'},
    {'x-amz-credential': `${process.env.AWS_ACCESS_KEY_ID}/20180820/us-east-2/s3/aws4_request`}
  ],
  Expires: 300  // 300 seconds
}

s3.createPresignedPost(params, (err, data) => {
  console.log(data);
});
```

Note that I've also tried using the key `ContentType` in the params.

And I'm wrapping tests around the upload processes that should not work. (asserting a validation failure) I'm finding that the content type is not enforced there as I can upload other file types with other content-type post params.

It's not clear to me whether the JS SDK manages the signing process for the user or if I need to do something special to get these various keys in the signature.

It's clear that I CAN do some of this, it's not clear if I NEED to. Not sure if the library is supposed to be handling that for me.

https://github.com/aws/aws-sdk-js/blob/5d868465426a66c9bdae1f3f0bb67aeca5a5825f/lib/signers/v4.js#L192 also seems to be of interest.  Is it the intention that we don't want to be using `content-type` as part of validation?  It seems like others are able to use validation based on `content-type` and I'm curious if it's possible to do with the js sdk.  Note that I am also having issues with `content-length`, which is also mentioned in that list.

TLDR; What do I need to do to activate content-type validation with s3 pre-signed POST's using the js sdk?

Any help would be greatly appreciated.

I've described my question here as well:
https://stackoverflow.com/questions/51935747/content-type-not-enforced-in-s3-pre-signed-post

Similar issues:
https://github.com/aws/aws-sdk-js/issues/1703

**Comment by srchase** [comment]
@tglines,

How are you currently using the response from createPresignedPost?

If you append the key/value pairs to formData before submitting, and attempt to upload a file with a different filetype than what was set in createPresignedPost, you'll get back a 403 response that includes the following:

<Code>AccessDenied</Code>
<Message>Invalid according to Policy: Policy Condition failed: [\"eq\", \"$Content-Type\", \"image/png\"]</Message>

**Comment by srchase** [comment]
@tglines 

Were you able to get this resolved?  I was hoping to get some additional details your use case.

**Comment by srchase** [comment]
Closing out this issue.  Let us know if you have anything additional to add.
